pkg/apis/v1alpha1/nsxserviceaccount_types.go

```diff
@@ -46,18 +46,22 @@ const (
 	NSXServiceAccountPhaseRealized   NSXServiceAccountPhase = "realized"
 	NSXServiceAccountPhaseInProgress NSXServiceAccountPhase = "inProgress"
 	NSXServiceAccountPhaseFailed     NSXServiceAccountPhase = "failed"
+	ConditionTypeRealized            string                 = "Realized"
 )
 
 // NSXServiceAccountStatus defines the observed state of NSXServiceAccount
 type NSXServiceAccountStatus struct {
-	Phase          NSXServiceAccountPhase `json:"phase,omitempty"`
-	Reason         string                 `json:"reason,omitempty"`
-	VPCPath        string                 `json:"vpcPath,omitempty"`
-	NSXManagers    []string               `json:"nsxManagers,omitempty"`
-	ProxyEndpoints NSXProxyEndpoint       `json:"proxyEndpoints,omitempty"`
-	ClusterID      string                 `json:"clusterID,omitempty"`
-	ClusterName    string                 `json:"clusterName,omitempty"`
-	Secrets        []NSXSecret            `json:"secrets,omitempty"`
+	// Deprecated: Use Condition instead.
+	Phase NSXServiceAccountPhase `json:"phase,omitempty"`
+	// Deprecated: Use Condition instead.
```

Comment:
Is there a way to mark Phase and Reason deprecated in CRD, too?

After some googling, I found the following example.

```
// +kubebuilder:deprecatedversion:warning="example.com/v1alpha1 CronTab is deprecated" 
YourField FieldType `json:"cronTab,omitempty"`
```

Reply:
I added this comment but found it make no effect on the generated code/yaml and the IDE highlights.